bee-node/src/default_plugins/mps.rs

```diff
@@ -15,37 +15,24 @@ use bee_common_ext::event::Bus;
 use bee_protocol::event::TpsMetricsUpdated;
 
 use log::info;
+use async_trait::async_trait;
 
-use std::{convert::Infallible, sync::Arc};
+use std::convert::Infallible;
 
-fn tps(metrics: &TpsMetricsUpdated) {
-    info!(
-        "incoming {} new {} known {} invalid {} outgoing {}",
-        metrics.incoming, metrics.new, metrics.known, metrics.invalid, metrics.outgoing
-    );
-}
-
-pub(crate) struct TpsPlugin {}
-
-impl TpsPlugin {
-    pub(crate) fn new() -> Self {
-        Self {}
-    }
-}
+pub struct Mps;
 
-impl Plugin for TpsPlugin {
+#[async_trait]
+impl Plugin for Mps {
+    type Config = ();
     type Error = Infallible;
 
-    fn name(&self) -> &str {
-        "tps"
-    }
-
-    fn init(&mut self, bus: Arc<Bus>) -> Result<(), Self::Error> {
-        bus.add_listener::<(), _, _>(tps);
-        Ok(())
-    }
-
-    fn start(&mut self) -> Result<(), Self::Error> {
-        Ok(())
+    async fn start(_: Self::Config, bus: &Bus<'_>) -> Result<Self, Self::Error> {
+        bus.add_listener::<(), TpsMetricsUpdated, _>(|metrics| {
```

Comment:
Weren't we supposed to pass the worker type in here ?

Reply:
This was part of my change to how the bus was accessed. I think it needs more thought though (since the actual worker type is non-obvious).